src/mono/wasm/Wasm.Build.Tests/BuildTestBase.cs

```diff
@@ -644,7 +648,22 @@ static void AssertRuntimePackPath(string buildOutput, string targetFramework)
             });
 
             AssertFilesExist(bundleDir, new[] { "run-v8.sh" }, expectToExist: hasV8Script);
-            AssertFilesExist(bundleDir, new[] { "icudt.dat" }, expectToExist: hasIcudt);
+            switch (globalizationMode)
+            {
+                case GlobalizationMode.Invariant:
+                    AssertFilesExist(bundleDir, new[] { "icudt.dat" }, expectToExist: false);
+                    break;
+                case GlobalizationMode.FullIcu:
+                    AssertFilesExist(bundleDir, new[] { "icudt.dat" }, expectToExist: true);
+                    break;
+                case GlobalizationMode.PredefinedIcu:
+                    if (string.IsNullOrEmpty(predefinedIcudt))
+                        throw new ArgumentException("WasmBuildTest is invalid, value for predefinedIcudt is required when GlobalizationMode=PredefinedIcu.");
+                    AssertFilesExist(bundleDir, new[] { predefinedIcudt }, expectToExist: true);
+                    break;
+                case GlobalizationMode.AutomaticIcu:
+                    throw new ArgumentException($"Running WBT in GlobalizationMode=AutomaticIcu is not allowed. In this mode icu loading depends on the locale of environment where the test is running and no check if globalization data was loaded is performed.");
```

Comment:
I'm not sure I understand this. What is using this?

Reply:
It's a warning for future developers not to user the automatic icu shard selection mode (based on the browser's locale) in the tests because it's not deterministic. We should run tests either in invariant, full icu mode or pass the exact name of icu shard.

Reply:
We want to test the *default* case for most (non-globalization related) of the tests, so explicitly set a locale for those. And then this mode shouldn't be needed at all.